spec/tapioca/compilers/dsl/mixed_in_class_attributes_spec.rb

```diff
@@ -0,0 +1,130 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "spec_helper"
+
+class Tapioca::Compilers::Dsl::MixedInClassAttributesSpec < DslSpec
+  before do
+    require "active_support/core_ext/class/attribute"
+    require "active_support/concern"
+  end
+
+  describe("#gather_constants") do
+    it("gathers modules that respond to class_attribute") do
+      add_ruby_file("file.rb", <<~RUBY)
+        module ManualIncluded
+          def self.included(base); end
+        end
+
+        module Concern
+          extend ActiveSupport::Concern
+        end
+
+        module SomeOtherModule
+        end
+      RUBY
+
+      assert_includes(gathered_constants, "ManualIncluded")
+      assert_includes(gathered_constants, "Concern")
+      refute_includes(gathered_constants, "SomeOtherModule")
```

Comment:
I prefer checking for strict array equality here since that would catch it if some other module were to sneak into this list. Moreover `gathered_constants` does a built-in sort to make that easy.

```suggestion
      assert_equal(["ManualIncluded", "Concern"], gathered_constants)
```